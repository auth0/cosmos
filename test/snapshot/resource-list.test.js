import React from 'react'
import renderer from 'react-test-renderer'
import { ResourceList } from '@auth0/cosmos'
import 'jest-styled-components'

describe('ResourceList', () => {
  it('renders with automation attribute', () => {
    const tree = renderer
      .create(
        <ResourceList
          items={[
            {
              title: 'Item 1',
              subtitle: 'Native'
            }
          ]}
          actions={[]}
          onItemClick={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
