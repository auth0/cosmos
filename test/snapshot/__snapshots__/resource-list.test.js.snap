// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceList renders with automation attribute 1`] = `
.c0 {
  margin: 32px 0;
  padding: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  border-top: 1px solid #DDD;
  padding: 16px 8px;
  cursor: pointer;
}

.c1:hover {
  background: #FAFAFA;
}

.c2 {
  -webkit-flex-basis: 40%;
  -ms-flex-preferred-size: 40%;
  flex-basis: 40%;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .sc-bxivhb {
  margin-right: 16px;
}

.c3 {
  margin-top: 8px;
  display: block;
}

.c4 {
  font-size: 12px;
  color: #757575;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-transform: uppercase;
}

.c5 {
  -webkit-flex-basis: 40%;
  -ms-flex-preferred-size: 40%;
  flex-basis: 40%;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  -webkit-flex-basis: 20%;
  -ms-flex-preferred-size: 20%;
  flex-basis: 20%;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c7 .cosmos-tooltip {
  left: 58%;
}

.c7 .sc-gqjmRU {
  margin-left: 8px;
}

<ul
  className="c0"
  data-cosmos-key="resource-list"
>
  <li
    className="c1"
    data-cosmos-key="resource-list.item"
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div>
        Item 1
        <span
          className="c3 c4"
        >
          Native
        </span>
      </div>
    </div>
    <div
      className="c5"
    />
    <div
      className="c6"
    >
      <div
        className="c7"
        data-cosmos-key="button-group"
      />
    </div>
  </li>
</ul>
`;
